sqlalchemy-stubs/util/_collections.pyi

```diff
@@ -1,101 +1,142 @@
+    def items(self) -> List[Tuple[_KT, _VT]]: ...  # type: ignore
+    if sys.version_info < (3, 0):
+        def itervalues(self) -> Iterator[_VT]: ...
+        def iterkeys(self) -> Iterator[_KT]: ...
+        def iteritems(self) -> Iterator[Tuple[_KT, _VT]]: ...
+    def __setitem__(self, key: _KT, object: _VT) -> None: ...
+    def __delitem__(self, key: _KT) -> None: ...
+    @overload
+    def pop(self, key: _KT) -> _VT: ...
+    @overload
+    def pop(self, key: _KT, *default: Union[_VT, _T]) -> Union[_VT, _T]: ...
+    def popitem(self) -> Tuple[_KT, _VT]: ...
+
+_OS = TypeVar('_OS', bound=OrderedSet)
+
+class OrderedSet(Set[_T]):
+    def __init__(self, d: Optional[Iterable[_T]] = ...) -> None: ...
+    def add(self, element: _T) -> None: ...
+    def remove(self, element: _T) -> None: ...
+    def insert(self, pos: int, element: _T) -> None: ...
+    def discard(self, element: _T) -> None: ...
+    def clear(self) -> None: ...
+    def __getitem__(self, key: int) -> _T: ...
+    def __iter__(self) -> Iterator[_T]: ...
+    def __add__(self, other: Iterable[_T]) -> _OS: ...
+    def update(self: _OS, iterable: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __ior__(self: _OS, iterable: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def union(self: _OS, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __or__(self: _OS, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def intersection(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __and__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def symmetric_difference(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __xor__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def difference(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __sub__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def intersection_update(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __iand__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    # signature incompatible with supertype
+    def symmetric_difference_update(self, other: Iterable[_T]) -> _OS: ...  # type: ignore
+    def __ixor__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def difference_update(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
+    def __isub__(self, other: Iterable[_T]) -> _OS: ...  # type: ignore  # signature incompatible with supertype
```

Comment:
This is suspicious. I think this class essentially just mimics usual Python set semantics. I would double check behavior of these special methods with respect to subclassing, and with respect to other type in `other` (see `builtins.set`).

Reply:
It tries to mimic the set semantics, but it doesn't completely do it. For instance, `difference()` only takes one argument whereas `set.difference()` takes a variable number of others. I'll update the signatures to be more set-like, but some of them will still need to be ignored because of the differences.